### PR TITLE
[GH-954]:Fixed issue 'DM comment notifications not working'

### PR DESCRIPTION
**Pull request by Kshitij-Katiyar**
**Summary** 
- Fixed the issue that users were not getting DM notifications when they are mentioned in a Jira issue comment.
- Updated the code to replace the jira username or email address with mattermost username.

**Issue** 

- [Issue #954](https://github.com/mattermost/mattermost-plugin-jira/issues/954)

### CODE REVIEW

File: server/utils.go [hunk 1]
```diff
@@ -84,25 +84,25 @@ func (p *Plugin) CreateBotDMtoMMUserID(mattermostUserID, format string, args ...
...
-		jiraUserID := uname[len("accountid:"):]
-		mattermostUserID, err := p.userStore.LoadMattermostUserID(instanceID, jiraUserID)
+		mattermostUserID, err := p.userStore.LoadMattermostUserID(instanceID, jiraUserIDOrName)
 		if err != nil {
 			continue
 		}
-		c, err := p.userStore.LoadConnection(instanceID, mattermostUserID)
-		if err != nil {
+
+		user, appErr := p.API.GetUser(string(mattermostUserID))
```
Comment (mickmister) [inline server/utils.go:100]:
```suggestion
		user, err := p.client.User.Get(string(mattermostUserID))
```

File: server/webhook_http_test.go [hunk 2]
```diff
@@ -628,6 +628,9 @@ func TestWebhookHTTP(t *testing.T) {
 			}
 			model.ParseSlackAttachment(rPost, rAttachments)
 			api.On("CreatePost", mock.AnythingOfType("*model.Post")).Return(rPost, nil)
+			api.On("GetUser", mock.AnythingOfType("string")).Return(&model.User{
+				Username: "test-username",
+			}, nil)
```
Comment (mickmister) [inline server/webhook_http_test.go:633]: Curious why no tests are failing with the change in this PR. Can we add a test to assert that the username replace is working correctly?

File: server/webhook_http_test.go [hunk 3]
```diff
@@ -600,6 +600,20 @@ func TestWebhookHTTP(t *testing.T) {
 			ExpectedIgnored: true,
 			CurrentInstance: false,
 		},
+		"CLOUD comment created - user mentioned": {
+			Request:                 testWebhookRequest("webhook-cloud-comment-created-mention-user.json"),
+			ExpectedSlackAttachment: true,
+			ExpectedHeadline:        "Test User **commented** on story [TES-41: Unit test summary 1](https://some-instance-test.atlassian.net/browse/TES-41)",
+			ExpectedText:            "> Added a comment with mentioned user [~test-mm-username]",
```
Comment (mickmister) [inline server/webhook_http_test.go:607]: I think we want this to be `@test-mm-username` right? To have a working MM user mention

Comment (mickmister) [reply server/webhook_http_test.go:607]: I don't want to block on the main purpose of this PR, but if we're going to change the mention functionality here, we should make it work as we intended in our PR discussion

**Review by mickmister — APPROVED** [review]
LGTM :+1:

**Comment by mickmister** [comment]
@DHaussermann Heads up that this is a customer reported issue. LGTM ready for QA review

**Review by AayushChaudhary0001 — APPROVED** [review]

**Pull request merged**